Mathlib/Data/MvPolynomial/Division.lean

```diff
@@ -117,6 +126,43 @@ theorem coeff_modMonomial_of_le {s' s : σ →₀ ℕ} (x : MvPolynomial σ R) (
   x.modOf_apply_of_exists_add _ _ <| exists_add_of_le h
 #align mv_polynomial.coeff_mod_monomial_of_le MvPolynomial.coeff_modMonomial_of_le
 
+lemma neg_modMonomial (s : σ →₀ ℕ) (x : MvPolynomial σ R') :
+    (-x) %ᵐᵒⁿᵒᵐⁱᵃˡ s = -(x %ᵐᵒⁿᵒᵐⁱᵃˡ s) :=
+  x.neg_modOf _
+
+lemma add_modMonomial (s : σ →₀ ℕ) (x y : MvPolynomial σ R) :
+    (x + y) %ᵐᵒⁿᵒᵐⁱᵃˡ s = (x %ᵐᵒⁿᵒᵐⁱᵃˡ s) + (y %ᵐᵒⁿᵒᵐⁱᵃˡ s) :=
+  x.add_modOf _ _
+
+lemma sub_modMonomial (s : σ →₀ ℕ) (x y : MvPolynomial σ R') :
+    (x - y) %ᵐᵒⁿᵒᵐⁱᵃˡ s = (x %ᵐᵒⁿᵒᵐⁱᵃˡ s) - (y %ᵐᵒⁿᵒᵐⁱᵃˡ s) :=
+  x.sub_modOf _ _
+
+lemma modMonomial_idem (s : σ →₀ ℕ) (x : MvPolynomial σ R) :
+    (x %ᵐᵒⁿᵒᵐⁱᵃˡ s) %ᵐᵒⁿᵒᵐⁱᵃˡ s = x %ᵐᵒⁿᵒᵐⁱᵃˡ s :=
+  x.modOf_idem _
+
+lemma mul_divMonomial (x y : MvPolynomial σ R) (s : σ →₀ ℕ) :
+    (x * y) /ᵐᵒⁿᵒᵐⁱᵃˡ s =
+    monomial s 1 * (x /ᵐᵒⁿᵒᵐⁱᵃˡ s) * (y /ᵐᵒⁿᵒᵐⁱᵃˡ s) +
+    (x /ᵐᵒⁿᵒᵐⁱᵃˡ s) * (y %ᵐᵒⁿᵒᵐⁱᵃˡ s) +
+    (x %ᵐᵒⁿᵒᵐⁱᵃˡ s) * (y /ᵐᵒⁿᵒᵐⁱᵃˡ s) +
+    ((x %ᵐᵒⁿᵒᵐⁱᵃˡ s) * (y %ᵐᵒⁿᵒᵐⁱᵃˡ s)) /ᵐᵒⁿᵒᵐⁱᵃˡ s :=
+  x.mul_divOf y s
+
+lemma mul_modMonomial (s : σ →₀ ℕ) (x y : MvPolynomial σ R) :
+    (x * y) %ᵐᵒⁿᵒᵐⁱᵃˡ s = ((x %ᵐᵒⁿᵒᵐⁱᵃˡ s) * (y %ᵐᵒⁿᵒᵐⁱᵃˡ s)) %ᵐᵒⁿᵒᵐⁱᵃˡ s :=
+  x.mul_modOf _ _
+
+lemma mul_modMonomial_finsupp_single (i : σ) (x y : MvPolynomial σ R) :
+    (x * y) %ᵐᵒⁿᵒᵐⁱᵃˡ Finsupp.single i 1 =
+    (x %ᵐᵒⁿᵒᵐⁱᵃˡ Finsupp.single i 1) * (y %ᵐᵒⁿᵒᵐⁱᵃˡ Finsupp.single i 1) := by
+  refine AddMonoidAlgebra.mul_modOf_of_minimal _ _ _ fun a b ha hb h => ?_
+  rw [← le_iff_exists_add, Finsupp.single_le_iff] at ha hb h
+  rw [not_le, Nat.lt_one_iff] at ha hb
+  rw [Finsupp.add_apply, ha, hb, zero_add] at h
+  cases h
```

Comment:
@jjaassoonn: I substantially golfed this in bd4602f88f07fe6b1a3e5bf11ffee56bb84313c2

Reply:
Nice! Thanks